app/models/miq_expression/relative_datetime.rb

```diff
@@ -58,4 +63,11 @@ def self.normalize(rel_time, tz, mode = "beginning")
       ts
     end
   end
+
+  def self.normalize_date_range(is_date, start_val, end_val, tz)
+    value = normalize_date(is_date, start_val, tz, "beginning")
+    return value if is_date && start_val == end_val && !relative?(start_val)
```

Comment:
before refactoring I didn't see that. after, yes it looks obvious.

what a fun journey with these methods.

I'll change